macro/macroutils.py

```diff
@@ -247,13 +252,51 @@ def load_repo_devel_job_data(repo_name, distro=None):
     """
     return _load_manifest_file(repo_devel_job_data_file(repo_name, distro), repo_name, 'devel job data for repo')
 
-def load_repo_haros_data(repo_name, distro=None):
+def get_repo_reports_data(repo_name, distro=None):
     """
-    Load haros_data.yaml properties into dictionary for repo
-    @return: manifest properties dictionary
+    Search the rosdoc test restults / report directory for available reports
+    @return List of found reports, each report a dict containing "name" and "urls"
     @raise UtilException: if unable to load. Text of error message is human-readable
     """
-    return _load_manifest_file(repo_haros_data_file(repo_name, distro), repo_name, 'HAROS data for repo')
+    reportsdir = repo_report_dir(repo_name, distro)
+    if not os.path.exists(reportsdir):
+        return []
+    #
+    reportsdir_contents = os.listdir(reportsdir)
+    reports = []
+    # build the base URL for links
+    if distro:
+        reportsurl = doc_url + distro + '/devel_jobs/' + repo_name + "/test_results/"
+    else:
+        reportsurl = doc_url + 'devel_jobs/' + repo_name + "/test_results/"
+    #
+    for report_dir in reportsdir_contents:
+        report_path = os.path.join(reportsdir,report_dir)
+        if not os.path.isdir(report_path):
+            continue
```

Comment:
Does this structure (one level of directories, then only files) need to be enforced? Why not just recursively consider all files under this base directory?

Reply:
Some test results like HAROS are webpages with a deep file structure. In these cases, showing links like `test_result/css/my_style.css` would be counter productive.